rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -32,58 +34,108 @@
+        co.rsk.core.Coin receiverBalance = track.getBalance(rskReceiver);
+        co.rsk.core.Coin expectedReceiverBalance = receiverBalance.add(co.rsk.core.Coin.fromBitcoin(BTC_TRANSFERRED));
 
-        PartialMerkleTree pmtWithTransactions = createValidPmtForTransactions(Collections.singletonList(bitcoinTransaction.getHash()), bridgeConstants.getBtcParams());
-        int btcBlockWithPmtHeight = bridgeConstants.getBtcHeightWhenPegoutTxIndexActivates() + bridgeConstants.getPegoutTxIndexGracePeriodInBtcBlocks();
-        int chainHeight = btcBlockWithPmtHeight + bridgeConstants.getBtc2RskMinimumAcceptableConfirmations();
+        // Act
+        registerBtcTransactionOrFail();
 
+        // Assert
         try {
-            recreateChainFromPmt(btcBlockStoreWithCache, chainHeight, pmtWithTransactions, btcBlockWithPmtHeight, bridgeConstants.getBtcParams());
-        } catch (BlockStoreException e) {
+            Optional<Long> heightIfBtcTxHashIsAlreadyProcessed = bridgeStorageProvider.getHeightIfBtcTxhashIsAlreadyProcessed(bitcoinTransaction.getHash());
+            assertTrue(heightIfBtcTxHashIsAlreadyProcessed.isPresent());
+            assertEquals(rskExecutionBlock.getNumber(), heightIfBtcTxHashIsAlreadyProcessed.get());
+        } catch (IOException e) {
             fail(e.getMessage());
         }
 
-        bridgeStorageProvider.save();
+        assertEquals(expectedFederationUtxos, federationSupport.getActiveFederationBtcUTXOs());
+        assertEquals(expectedReceiverBalance, repository.getBalance(rskReceiver));
 
-        BridgeSupport bridgeSupport = getBridgeSupport(bridgeStorageProvider, activationConfig, federationSupport, feePerKbSupport, rskExecutionBlock, btcBlockStoreFactory, track, btcLockSenderProvider);
+    }
 
-        Transaction rskTx = TransactionUtils.createTransaction();
-        org.ethereum.crypto.ECKey key = org.ethereum.crypto.ECKey.fromPublicOnly(btcPublicKey.getPubKey());
+    @Test
+    void whenRegisterARepeatedLegacyBtcTransaction_shouldNotPerformAnyChange() {
+        // Arrange
+        registerBtcTransactionOrFail();
 
-        RskAddress receiver = new RskAddress(key.getAddress());
-        co.rsk.core.Coin receiverBalance = track.getBalance(receiver);
-        co.rsk.core.Coin expectedReceiverBalance = receiverBalance.add(co.rsk.core.Coin.fromBitcoin(btcTransferred));
+        RskAddress receiverBalance = new RskAddress(ecKey.getAddress());
+        co.rsk.core.Coin expectedReceiverBalance = track.getBalance(receiverBalance);
+        List<UTXO> expectedFederationUTXOs = federationSupport.getActiveFederationBtcUTXOs();
 
         // Act
+        registerBtcTransactionOrFail();
+
+        // Assert
+        assertEquals(expectedFederationUTXOs, federationSupport.getActiveFederationBtcUTXOs());
+        assertEquals(expectedReceiverBalance, repository.getBalance(receiverBalance));
+    }
+
+    private void registerBtcTransactionOrFail() {
```

Comment:
Avoid these try-catch's. If `bridgeSupport.registerBtcTransaction` throws an exception just declare it in the method signature.